### ISSUE TITLE
`transformPage`

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/kit#3511]
### Describe the problem

Until recently, the way to manipulate HTML was to do something like this in `handle`:

```js
export async function handle({ request, resolve }) {
  const rendered = await resolve(request);

  if (rendered.headers['content-type'] === 'text/html') {
    rendered.body = rendered.body.replace(/old/g, 'new');
  }

  return rendered;
}
```

With #3384, we're now dealing with `Response` objects meaning that you'd now do this:

```js
export async function handle({ event, resolve }) {
  let response = await resolve(event);

  if (response.headers.get('content-type') === 'text/html') {
    const body = await response.text();
    response = new Response(body.replace(/old/g, 'new'), {
      status: response.status,
      headers: response.headers
    });
  }

  return response;
}
```

For all the advantages brought by using `Request` and `Response`, this does not feel like an improvement. Verbosity aside, doing `await response.text()` and creating a new response feels like a lot of overhead.

### Describe the proposed solution

We could do this:

```js
export async function handle({ event, resolve }) {
  return await resolve(event, {
    transformPage: ({ html }) => html.replace(/old/g, 'new')
  });
}
```

Then this line... https://github.com/sveltejs/kit/blob/0cca67063a4f38806eb4428b76548b8d948035ca/packages/kit/src/runtime/server/page/render.js#L218 ...would become something like 

```js
const html = options.transformPage(options.template({ head, body, assets }));
```

### Alternatives considered

The alternative is to do nothing. It depends on whether we consider manipulating the HTML response to be 'supported' — if it's a rarely-used hack, perhaps it's not necessary, but if we consider it to be a valid and common use case then the API probably pays for itself.

Minification is one use case; injecting the correct `<html lang="...">` attribute is another (albeit one that Kit will eventually have full support for anyway). I've seen it be used for other stuff — here's a snippet from the NYT covid tracker that allows multiple Kit apps to exist on the same page...

```js
/** @type {import('@sveltejs/kit').Handle} */
export async function handle({ request, resolve }) {
  const rendered = await resolve(request);

  if (typeof rendered.body === 'string') {
    rendered.body = rendered.body
      .replace(/start\({([\s\S]+?)}\);/, (match, $1) => {
        const opts = $1.replace(/target: .+?,/, 'target,');
        return `const target = document.querySelector('#__covidtracker__');\n\t\t\ttarget.removeAttribute('id');\n\t\t\tstart({${opts}});`;
      })
      .replace(/__covidtracker__/g, `svelte-${hash(request.path)}`);
  }

  return rendered;
}
```

...and there are doubtless other uses we can't anticipate. So I vote for its inclusion, but others might feel differently.

### Importance

nice to have

**Comment by ebeloded** [comment]
Thank you for pointing out how it can be done after #3384!

Yes, please. This would allow for a better way to handle [templating of app.html](https://github.com/sveltejs/kit/issues/3091).

Here's what it would look like to allow `ejs` syntax templating in `app.html`:

```ts
export async function handle({ event, resolve }) {
  return await resolve(event, {
    transformPage: ({ html }) => ejs.render(html, { foo: 'bar' }))
  });
}
```

**Comment by rodoch** [comment]
I also think this would be useful. I have to do these kinds of operations in almost every SK app I build and this syntax is concise while making the intent clearer.

**Comment by itssumitrai** [comment]
Definitely very useful for my use case as well. Thanks 👍
